llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -138,10 +141,16 @@ def _get_site_id_with_host_name(self, access_token, sharepoint_site_name) -> str
         if hasattr(self, "_site_id_with_host_name"):
             return self._site_id_with_host_name
 
-        site_information_endpoint = f"https://graph.microsoft.com/v1.0/sites"
-
         self._authorization_headers = {"Authorization": f"Bearer {access_token}"}
 
+        if self.sharepoint_site_id:
+            sharepoint_side_id = self.sharepoint_site_id
```

Comment:
I think you don't need to assign `sharepoint_site_id` (there's a typo BTW), you can just use `self.sharepoint_site_id` at line 149